common/src/main/java/org/conscrypt/CacheableSession.java

```diff
@@ -0,0 +1,389 @@
+package org.conscrypt;
+
+import static org.conscrypt.SSLUtils.SessionType.OPEN_SSL_WITH_OCSP;
+import static org.conscrypt.SSLUtils.SessionType.OPEN_SSL_WITH_TLS_SCT;
+import static org.conscrypt.SSLUtils.SessionType.isSupportedType;
+
+import java.io.ByteArrayOutputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.nio.BufferUnderflowException;
+import java.nio.ByteBuffer;
+import java.security.Principal;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateEncodingException;
+import java.util.List;
+import javax.net.ssl.SSLException;
+import javax.net.ssl.SSLPeerUnverifiedException;
+import javax.net.ssl.SSLSession;
+import javax.net.ssl.SSLSessionContext;
+import javax.security.cert.X509Certificate;
+
+/**
+ * An fully established {@link SSLSession} that may be cached.
+ *
+ * <p/>It should be noted that many methods are not implemented due to the fact that
+ * BoringSSL's API only exposes things such as certificates and OCSP stapled responses on the
+ * SSL object, but not the SSL_SESSION. If the session is serialized (i.e. {@link #toBytes()}),
+ * BoringSSL will serialize all of the information so that when it is later deserialized
+ * (i.e. {@link #newInstance(byte[], String, int, AbstractSessionContext)}) and offered to
+ * another connection, it will have all of the information needed to resume the session.
+ */
+final class CacheableSession implements SSLSession {
```

Comment:
Yes, that seems like it would simplify this section of the code. It never escapes, so it should be a simple change correct?